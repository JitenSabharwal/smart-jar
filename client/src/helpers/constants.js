export const ADD_CONTAINER_REQUEST = 'ADD_CONTAINER_REQUEST'
export const ADD_CONTAINER_SUCCESS = 'ADD_CONTAINER_SUCCESS'
export const ADD_CONTAINER_FAILURE = 'ADD_CONTAINER_FAILURE'
export const UPDATE_CONTAINER_REQUEST = 'UPDATE_CONTAINER_REQUEST'
export const UPDATE_CONTAINER_SUCCESS = 'UPDATE_CONTAINER_SUCCESS'
export const UPDATE_CONTAINER_FAILURE = 'UPDATE_CONTAINER_FAILURE'
export const DELETE_CONTAINER = 'DELETE_CONTAINER'
export const DELETE_ALL_CONTAINERS = 'DELETE_ALL_CONTAINERS'
export const SELECT_CONTAINER = 'SELECT_CONTAINER'
export const CONTAINER_TABLE_HEADS = [
  { id: 'firstName', numeric: false, disablePadding: true, label: 'FirstName' },
  { id: 'lastName', numeric: false, disablePadding: false, label: 'LastName' },
  { id: 'hobbies', numeric: false, disablePadding: false, label: 'Hobbies' },
  { id: 'birthDate', numeric: false, disablePadding: false, label: 'BirthDate' },
  { id: 'url', numeric: false, disablePadding: false, label: 'Profile' },
]
export const DEFAULT_URL = '/uploads/default.jpeg'
export const MAX_LENGTH = 30
export const NAME_REGEX = /^[a-zA-Z ]{2,30}$/
