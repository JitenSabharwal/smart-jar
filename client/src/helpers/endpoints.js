export const ALL_CONTAINER_ENDPOINT = '/api/v1/containers'
export const ADD_CONTAINER_ENDPOINT = '/api/v1/containers/addContainer'
export const DELETE_CONTAINER_ENDPOINT = '/api/v1/containers/removeContainer'
export const DELETE_ALL_CONTAINER_ENDPOINT = '/api/v1/containers/removeAllContainers'
