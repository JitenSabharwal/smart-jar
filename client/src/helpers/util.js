import axios from 'axios'

/**
 * Function to do a prepare request headers for anything required
 */
export const perpareHeader = () => {
  const id = localStorage.getItem('id_token')
  const config = {
    headers: {
      'Authorization': id ? `Bearer ${id}` : '',
    },
  }
  return config
}

/**
 * Function to do a post Request
 * @param {string} url
 * @param {object} payload
 */
export const postRequest = (url, payload) => {
  const headers = perpareHeader()
  return axios.post(url, payload, headers)
}

/**
 * Function to do a get Request
 * @param {string} url
 * @param {object} payload
 */
export const getRequest = (url, payload) => {
  const headers = perpareHeader()
  return axios.get(url, headers)
}
/**
 * Funtion to filter data out
 * @param {array} data [{id: 'redquired', ...container properties}]
 * @param {*} val
 * @param {string} key If not _id any other prop
 */
export const findUser = (data, val, key = '_id') => {
  return data.filter(u => u[key] === val)
}

/**
 * Functiom to load set an initial state
 */
export const loadInitailState = (url) => {
  return new Promise((resolve, reject) => {
    getRequest(url)
      .then(resp => {
        const {success, data} = resp.data
        if (!success) {
          // If there was a problem, we want to
          return reject(resp)
        } else {
          // If login was successful, set the token in local storage
          const state = {
            container: {
              list: data || [],
            },
          }
          resolve(state)
        }
      })
      .catch(err => {
        reject(new Error('Error: ', err))
      })
  })
}

/**
 * Function returns -1, 0, 1
 *  depending on the match of the
 *  property orderBy
 *  of elements a and b
 * @param {object} a
 * @param {object} b
 * @param {string} orderBy
 */
function desc (a, b, orderBy) {
  if (b[orderBy] < a[orderBy]) {
    return -1
  }
  if (b[orderBy] > a[orderBy]) {
    return 1
  }
  return 0
}
/**
 * Function return a Function to sort elements
 * by orderBy
 * in the given order
 * @param {string} order 'desc/asc'
 * @param {string} orderBy '<string property of the sorting object>'
 */
export const getSorting = (order, orderBy) => {
  return order === 'desc' ? (a, b) => desc(a, b, orderBy) : (a, b) => -desc(a, b, orderBy)
}
