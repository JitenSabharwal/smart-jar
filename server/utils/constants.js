export const ALL_CONTAINER_ENDPOINT = '/api/v1/user'
export const ADD_CONTAINER_ENDPOINT = '/api/v1/user/addUser'
export const UPDATE_CONTAINER_ENDPOINT = '/api/v1/user/updateUser'
export const DELETE_CONTAINER_ENDPOINT = '/api/v1/user/removeUser'
export const DELETE_ALL_CONTAINER_ENDPOINT = '/api/v1/user/removeAllUsers'
