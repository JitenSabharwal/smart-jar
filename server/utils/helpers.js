const log = require('./logger/logger')
/**
 * Parse function to prepare output json
 * @param {object} obj {data: <object>, error:<boolean>, errorMsg: <String>}
 */
const parse = obj => {
  const result = {
    success: true,
  }
  if (obj.error) {
    result.success = false
    result.error = obj.error
  }
  result.data = obj.data
  return result
}

/**
 * Function creates output user object from the given data
 * @param {object} obj {
      itemName: String,
      quantity: String,
      timestamp: Date/String(DD/MM/YYYY),
    }
 */
const outputContainer = (arr) => {
  try {
    const containers = {
      itemName: arr[0].itemName,
      details: arr.map(a => ({quantity: a.quantity, timestamp: a.timestamp})).sort((a, b) => +new Date(b.timestamp) - +new Date(a.timestamp)),
    }
    return containers
  } catch (e) {
    log.error(e)
  }
  return {}
}

module.exports = {
  parse,
  outputContainer,
}
