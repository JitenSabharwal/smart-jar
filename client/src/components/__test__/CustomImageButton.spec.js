/* eslint-disable no-undef */
import React from 'react'
import Enzyme, {shallow} from 'enzyme'
import CustomImageButton from '../CustomImageButton.jsx'
import Adapter from 'enzyme-adapter-react-16'
import {DEFAULT_URL} from '../../helpers/constants'
const image = {
  url: DEFAULT_URL,
  title: 'Click to change',
  width: '100%',
}
const onClick = () => {}
Enzyme.configure({adapter: new Adapter()})
describe('CustomImageButton', () => {
  test('render', () => {
    const wrapper = shallow(<CustomImageButton image={image} onClick={onClick} />)
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})
