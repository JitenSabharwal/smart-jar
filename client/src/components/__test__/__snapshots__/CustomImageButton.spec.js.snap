// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomImageButton render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ButtonBases)
    image={
      Object {
        "title": "Click to change",
        "url": "/uploads/default.jpeg",
        "width": "100%",
      }
    }
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "focusVisible": "ButtonBases-focusVisible-3",
        "image": "ButtonBases-image-2",
        "imageBackdrop": "ButtonBases-imageBackdrop-6",
        "imageButton": "ButtonBases-imageButton-4",
        "imageMarked": "ButtonBases-imageMarked-8",
        "imageSrc": "ButtonBases-imageSrc-5",
        "imageTitle": "ButtonBases-imageTitle-7",
        "root": "ButtonBases-root-1",
      },
      "image": Object {
        "title": "Click to change",
        "url": "/uploads/default.jpeg",
        "width": "100%",
      },
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "focusVisible": "ButtonBases-focusVisible-3",
          "image": "ButtonBases-image-2",
          "imageBackdrop": "ButtonBases-imageBackdrop-6",
          "imageButton": "ButtonBases-imageButton-4",
          "imageMarked": "ButtonBases-imageMarked-8",
          "imageSrc": "ButtonBases-imageSrc-5",
          "imageTitle": "ButtonBases-imageTitle-7",
          "root": "ButtonBases-root-1",
        },
        "image": Object {
          "title": "Click to change",
          "url": "/uploads/default.jpeg",
          "width": "100%",
        },
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
